core/app/api/v0/c/[communitySlug]/site/[...ts-rest]/route.ts

```diff
@@ -271,6 +275,55 @@ const parseQueryWithQsMiddleware: RequestMiddleware = (req) => {
 const handler = createNextHandler(
 	siteApi,
 	{
+		auth: {
+			check: {
+				siteBuilder: async () => {
+					const { authorization, community, isSiteBuilderToken } =
+						await getAuthorization();
+
+					if (!isSiteBuilderToken) {
+						return {
+							status: 401,
+							body: {
+								ok: false,
+								code: "NON_SITE_BUILDER_TOKEN",
+								reason: "The supplied token is not a site builder token. Either something went wrong with the token generation or the token was intercepted by a third party.",
+							},
+						};
+					}
+
+					// TODO: enable again when you're less silly
+					// for (const permission of allPermissions) {
+					// 	const exists = authorization[permission.scope]?.[permission.accessType];
+					// 	if (permission.accessType !== "read" && exists) {
+					// 		return {
+					// 			status: 401,
+					// 			body: {
+					// 				ok: false,
+					// 				code: "HAS_WRITE_PERMISSIONS",
+					// 				reason: `Site builder token has ${permission.accessType} permissions for ${permission.scope}, which is not allowed. Please contact support.`,
+					// 			},
+					// 		};
+					// 	}
+					// 	if (permission.accessType === "read" && !exists) {
+					// 		return {
+					// 			status: 401,
+					// 			body: {
+					// 				ok: false,
+					// 				code: "HAS_NO_READ_PERMISSIONS",
+					// 				reason: `Site builder token has no read permissions for ${permission.scope}, which is required. Please contact support.`,
+					// 			},
+					// 		};
+					// 	}
+					// }
```

Comment:
this was just extremely cautious but ended up causing problems, ill just remove it and we just trust that the sitebuilding token is good